src/button/__tests__/internal.test.tsx

```diff
@@ -23,3 +29,65 @@ test('supports __iconClass property', () => {
   );
   expect(container.querySelector(`button .${styles.icon}`)).toHaveClass('example-class');
 });
+
+describe('Analytics', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockFunnelMetrics();
+  });
+
+  test('does not send any metrics when not in a funnel context', () => {
+    const { container } = render(<Button iconName="external" target="_blank" href="https://example.com" />);
+    createWrapper(container).findButton()!.click();
+
+    expect(FunnelMetrics.externalLinkInteracted).not.toHaveBeenCalled();
+  });
+
+  test('sends an externalLinkInteracted metric within a Funnel Context with iconName=external', () => {
```

Comment:
Could we maybe add a test for when `iconName="external"` but no `href` is present?